readthedocs/notifications/notification.py

```diff
@@ -84,3 +87,74 @@ def send(self):
         avoided.
         """
         send_notification(self.request, self)
+
+
+class SiteNotification(Notification):
+
+    """
+    Simple notification to show *only* on site messages.
+
+    ``success_message`` and ``failure_message`` can be a simple string or a
+    dictionary with different messages depending on the reason of the failure /
+    success. The message is selected by using ``message_key`` to get the proper
+    value.
+
+    The notification is tied to the ``user`` and it could be sticky, persistent
+    or normal --this depends on the ``success_level`` and ``failure_level``.
+
+    .. note::
+
+        ``send_email`` is forced to False to not send accidental emails when
+        only a simple site notification is needed.
+    """
+
+    send_email = False
+
+    success_message = None
+    failure_message = None
+
+    success_level = constants.SUCCESS_NON_PERSISTENT
+    failure_level = constants.ERROR_NON_PERSISTENT
+
+    def __init__(
+            self, user, success, message_key=None, context_object=None,
+            request=None):
+        self.object = context_object
+
+        self.user = user or request.user
+        # Fake the request in case the notification is instantiated from a place
+        # without access to the request object (Celery task, for example)
+        self.request = request or HttpRequest()
+        self.request.user = user
+
+        self.success = success
+        self.message_key = message_key
+        super(SiteNotification, self).__init__(context_object, request, user)
+
+    def get_message_level(self):
+        if self.success:
+            return self.success_level
+        return self.failure_level
+
+    def get_message(self, success):
+        if success:
+            message = self.success_message
+        else:
+            message = self.failure_message
+
+        if isinstance(message, dict) and self.message_key:
+            if self.message_key in message:
+                msg = message.get(self.message_key)
+            else:
+                raise KeyError(
+                    "Notification has no key '{}' for {} messages".format(
+                        self.message_key,
+                        'success' if self.success else 'failure',
+                    ),
+                )
+        else:
+            msg = message
+        return msg.format(**{self.context_object_name: self.object})
```

Comment:
Perhaps this needs the django-y `get_context_data` as well? We might need a way to pass in variables to the message string (for link URL, etc).

For instance, in the case above, we might have separate URLs to pass in to the messages:

```python
failure_message = {
    SOMETHING: _('Hey, <a href="{url_connect_account}">connect your account</a>'),
    SOMETHING_ELSE: _('Hey, <a href="{url_docs_permissions}">read the docs</a>'),
}

def get_context_data(self):
    context_data = {self.context_object_name: self.object}
    context_data['url_connect_account'] = reverse(..., project=self.project)
    context_data['url_docs_permissions'] = 'https://docs.readthedocs.io/..'
    return context_data
```